mypy/plugins/common.py

```diff
@@ -110,5 +110,5 @@ def _add_method(
     func._fullname = info.fullname() + '.' + name
     func.line = info.line
 
-    info.names[name] = SymbolTableNode(MDEF, func)
+    info.names[name] = SymbolTableNode(MDEF, func, plugin_generated=True)
```

Comment:
How about making `_add_method` a public and recommended way to add methods? I would at least remove the leading underscore. Another idea is to modify this function signature so that it accepts a list of strings (dependencies) and adds them instead of providing `ctx.api.add_plugin_dependency` (I would hide as more details from plugin writers as possible).

Reply:
Happy to drop the underscore from `_add_method`. I just realized that `attrs` doesn't use it. It should be easy to make it use `add_method` but I realize now that I don't understand why `attrs` works without setting `plugin_generated`!

I'm not sure if avoiding direct calls to `add_plugin_dependency` would really help here. The plugin writer still would need to collect all the wildcard dependencies from the MRO, which is pretty grungy. Having to plumb that out to the `add_method` call seems like it complicates rather than hiding complexity.

Reply:
I am OK with keeping `add_plugin_dependency` (unless @JukkaL has other ideas).

Reply:
(It didn't work, I was just missing an attrs test.)

Reply:
I think it's better to keep `add_plugin_dependency`. The plugin writer will need to add the necessary dependencies in either way, and in this case it's better to have two API functions that do one thing rather than having a single function that does a bunch of things. It's possible that we'll add a new method, such as `add_attribute`, that would also need to take a list of dependencies, and now we'd have duplicate functionality.